listenbrainz/listenstore/timescale_listenstore.py

```diff
@@ -666,6 +681,227 @@ def dump_listens(self, location, dump_id, start_time=datetime.utcfromtimestamp(0
+        current_listened_at = None
+        conn = timescale.engine.raw_connection()
+        with conn.cursor(cursor_factory=psycopg2.extras.RealDictCursor) as curs:
+            curs.execute(query, args)
+            while True:
+                t0 = time.monotonic()
+                written = 0
+                approx_size = 0
+                data = {
+                    'listened_at': [],
+                    'user_name': [],
+                    'artist_name': [],
+                    'artist_credit_id': [],
+                    'release_name': [],
+                    'release_mbid': [],
+                    'recording_name': [],
+                    'recording_mbid': [],
+                    'artist_credit_mbids': []
+                }
+                while True:
+                    result = curs.fetchone()
+                    if not result:
+                        break
+
+                    for col in data:
+                        if col == 'artist_credit_id':
+                            artist_credit_ids.add(result[col])
+
+                        if col == 'listened_at':
+                            current_listened_at = datetime.utcfromtimestamp(result['listened_at'])
+                            data[col].append(current_listened_at)
+                            approx_size += len(str(result[col]))
+                        elif col == 'artist_credit_mbids':
+                            approx_size += AVG_ARTIST_MBIDS_LEN
+                        else:
+                            data[col].append(result[col])
+                            approx_size += len(str(result[col]))
+
+                    written += 1
+                    listen_count += 1
+                    if approx_size > PARQUET_TARGET_SIZE:
+                        break
+
+                if written == 0:
+                    break
+
+                # Fetch artist mbids for each artist_credit_id and then insert into data
+                # If an ac id is not found, zero out the other MBIDs since the underlying data changed
+                ac_mapping = self._fetch_artist_MBIDs_from_artist_credits(artist_credit_ids)
+                for i, row in enumerate(data['artist_credit_id']):
+                    if row is None:
+                        data['artist_credit_mbids'].append(None)
+                    elif row not in ac_mapping:
+                        data['artist_credit_mbids'].append(None)
+                        data['release_mbid'][i] = None
+                        data['recording_mbid'][i] = None
+                        data['artist_credit_id'][i] = None
+                    else:
+                        data['artist_credit_mbids'].append(ac_mapping[row])
+
+                filename = os.path.join(temp_dir, "%d.parquet" % parquet_file_id)
+
+                # Create a pandas dataframe, then write that to a parquet files
+                df = pd.DataFrame(data, dtype=object)
+                table = pa.Table.from_pandas(df) 
```

Comment:
I encountered a strange issue while generating some parquet files for tests. Due to some reason, pandas was adding an extra index column causing a schema mismatch in Spark. The issue got fixed when I added, `preserve_index=False`. (https://github.com/apache/arrow/issues/2244)

I haven't faced this issue when working with the parquet files generated by the dumps code but should we add this kwarg here as well to be on the safer side?

Reply:
Done.